pkg/repository/credentials_repository.go

```diff
@@ -13,30 +19,159 @@ func NewCredentialsRepository(db *storage.Redis) *CredentialsRepository {
 	return &CredentialsRepository{db: db}
 }
 
-func (cred *CredentialsRepository) SaveCredential(credential string) error {
+func (cred *CredentialsRepository) SaveCredential(credential string, address string) error {
 	ctx := context.Background()
 	// Add leaf to redis set (for fast membership check)
-	if err := cred.db.Client.SAdd(ctx, "merkle:credentials:set", credential).Err(); err != nil {
+	if err := cred.db.Client.SAdd(ctx, fmt.Sprintf("merkle:%s:credentials:set", address), credential).Err(); err != nil {
 		return err
 	}
 
 	// Add leaf to redis list (for ordered retrieval)
-	if err := cred.db.Client.RPush(ctx, "merkle:credentials:list", credential).Err(); err != nil {
+	if err := cred.db.Client.RPush(ctx, fmt.Sprintf("merkle:%s:credentials:set", address), credential).Err(); err != nil {
```

Comment:
Why are you creating a different key for each user? You won't be able to get all leaves afterwards.